### PR TITLE
Implement World1

### LINKED ISSUE #146: Implement World1 model
In the [MIT memorandum D-1348](https://dome.mit.edu/handle/1721.3/189645), "A World Dynamics Model: Introductory Exercise", Jay Forrester created the first world dynamics model.
As a matter of completeness and historical praise, we should also add World1 into [WorldDynamics.jl](https://github.com/natema/WorldDynamics.jl)

### PR DESCRIPTION
**Pull request by paulobruno**
This PR contains the corresponding code for the original World1 model. With it, we can generate all World1 figures, including adaptive scenarios.
Fixes #146

### CODE REVIEW

File: src/World1/world1/plots.jl [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+end
+
+function fig_5(; kwargs...)
+    new_params = copy(_params)
+    new_params[:brn] = 0.035
+
+    new_inits = copy(_inits)
+    new_inits[:ciaf] = 0.3
+
+    sol = solve(standard_run(params=new_params), (1900, 2100))
```
Comment (natema) [inline src/World1/world1/plots.jl:268]: The initializations were missing from the arguments:
```suggestion
    sol = solve(standard_run(inits=new_inits, params=new_params), (1900, 2100))
```
After the above fix, output seems equal to the figure in the book.

File: src/World1/world1/plots.jl [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+end
+
+function fig_6(; kwargs...)
+    new_params = copy(_params)
+    new_params[:brn] = 0.032
+
+    new_inits = copy(_inits)
+    new_inits[:ciaf] = 0.3
+
+    sol = solve(standard_run(params=new_params), (1900, 2100))
```
Comment (natema) [inline src/World1/world1/plots.jl:280]: The initializations were missing from the arguments:
```suggestion
    sol = solve(standard_run(inits=new_inits, params=new_params), (1900, 2100))
```
After the above fix, output seems equal to the figure in the book.

File: src/World1/world1/plots.jl [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+
+    plotvariables(sol, (t, 1900, 2000), _variables_6(); title="W1-7/5-6", showaxis=false, showlegend=true, kwargs...)
+end
+
+function fig_7(; kwargs...)
```
Comment (natema) [inline src/World1/world1/plots.jl:285]: :heavy_check_mark: output seems equal to the figure in the book.

File: src/World1/world1/plots.jl [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+    new_params = copy(_params)
+    new_params[:pols] = 10e9
+
+    sol = solve(standard_run(params=new_params), (1900, 2100))
+
+    plotvariables(sol, (t, 1900, 2100), _variables_7(); title="W1-7/5-7", showaxis=false, showlegend=true, kwargs...)
+end
+
+function fig_8(; kwargs...)
```
Comment (natema) [inline src/World1/world1/plots.jl:294]: :heavy_check_mark: output seems equal to the figure in the book.

File: src/World1/world1/plots.jl [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+    new_params = copy(_params)
+    new_params[:pdn] = 26.5
+
+    sol = solve(standard_run(params=new_params), (1900, 2100))
+
+    plotvariables(sol, (t, 1900, 2100), _variables_8(); title="W1-7/5-8", showaxis=false, showlegend=true, kwargs...)
+end
+
+function fig_9(; kwargs...)
```
Comment (natema) [inline src/World1/world1/plots.jl:303]: :heavy_check_mark: output seems equal to the figure in the book.

File: src/World1/world1/plots.jl [hunk 6]
```diff
@@ -0,0 +1,359 @@
...
+
+    new_tables = copy(_tables)
+    new_tables[:cipc] = (0.02, 0.05, 0.1, 0.14, 0.16, 0.18)
+
+    sol = solve(standard_run(params=new_params, tables=new_tables), (1900, 2100))
+
+    plotvariables(sol, (t, 1900, 2100), _variables_9(); title="W1-7/5-9", showaxis=false, showlegend=true, kwargs...)
+end
+
+function fig_10(; kwargs...)
```
Comment (natema) [inline src/World1/world1/plots.jl:315]: :heavy_check_mark: output seems equal to the figure in the book.

File: src/World1/world1/plots.jl [hunk 7]
```diff
@@ -0,0 +1,359 @@
...
+        (w1.ql,   0,     20,    "Quality of life"),
+        (w1.ciaf, 0.25,  0.65,  "Capital in agriculture"),
+        (w1.nr,   500e9, 900e9, "Natural resources"),
+    ]
+
+    return variables
+end
+
+
+fig_std(; kwargs...) = plotvariables(standardrunsolution(), (t, 1900, 2100), _variables_std(); title="WORLD1-STD", showaxis=false, showlegend=true, kwargs...)
```
Comment (natema) [inline src/World1/world1/plots.jl:217]: :heavy_check_mark: output seems equal to the figure in the book.

File: src/World1/world1/plots.jl [hunk 8]
```diff
@@ -0,0 +1,359 @@
...
+    new_tables = copy(_tables)
+    new_tables[:cipc] = (0.02, 0.05, 0.1, 0.14, 0.16, 0.18)
+    new_tables[:polat] = (0.5, 1.0, 1.6, 2.4, 3.2, 4.0, 4.8)
+
+    sol = solve(standard_run(params=new_params, tables=new_tables), (1900, 2100))
+
+    plotvariables(sol, (t, 1900, 2100), _variables_10(); title="W1-7/5-10", showaxis=false, showlegend=true, kwargs...)
+end
+
+function fig_11(; kwargs...)
```
Comment (natema) [inline src/World1/world1/plots.jl:328]: The generated plot is not included in the book, but the modifications are as described on page 55.

File: src/World1/world1/plots.jl [hunk 9]
```diff
@@ -0,0 +1,359 @@
...
+
+    new_inits = copy(_inits)
+    new_inits[:ciaf] = 0.3
+
+    sol = solve(standard_run(params=new_params, inits=new_inits, tables=new_tables), (1900, 2100))
+
+    plotvariables(sol, (t, 1900, 2100), _variables_10(); title="W1-7/5-11", showaxis=false, showlegend=true, kwargs...)
+end
+
+function fig_12(; kwargs...)
```
Comment (natema) [inline src/World1/world1/plots.jl:344]: As `fig_11`, this is also omitted from the book, and the required changes are as indicated on page 55.

**Review by natema — CHANGES_REQUESTED** [review]

File: src/World1/world1/plots.jl [hunk 10]
```diff
@@ -0,0 +1,359 @@
...
+
+    new_inits = copy(_inits)
+    new_inits[:ciaf] = 0.25
+
+    sol = solve(standard_run(inits=new_inits, params=new_params), (1900, 2100))
+
+    plotvariables(sol, (t, 1900, 2100), _variables_2(); title="W1-7/5-2", showaxis=false, showlegend=true, kwargs...)
+end
+
+function fig_3(; kwargs...)
```
[the remark this answers is not in the document]
Comment (paulobruno) [reply src/World1/world1/plots.jl:240]: Well observed. I hadn't noticed this.

**Pull request merged**